src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -487,16 +486,14 @@ public StorageService()
         // use dedicated executor for handling JMX notifications
         super(JMXBroadcastExecutor.executor);
 
-        jmxObjectName = "org.apache.cassandra.db:type=StorageService";
-
         sstablesTracker = new SSTablesGlobalTracker(DatabaseDescriptor.getSelectedSSTableFormat());
     }
 
-    public void registerMBeans()
+    protected void registerMBeans()
     {
-        logger.info("Initializing storage service mbean");
-        MBeanWrapper.instance.registerMBean(this, jmxObjectName);
-        MBeanWrapper.instance.registerMBean(StreamManager.instance, StreamManager.OBJECT_NAME);
+        logger.debug("Initializing storage service mbean");
+        MBeanWrapper.registerIfNotRegistered(this, MBEAN_NAME);
```

Comment:
I would go for `maybeRegister`. This "maybe" pattern is prevalent around the codebase afaict.